src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -108,29 +109,14 @@ public new void RemoveMetadataReference(string referencePath)
         public void AddProjectReference(IWorkspaceProjectContext project, MetadataReferenceProperties properties)
         {
             var abstractProject = GetAbstractProject(project);
-
-            // AbstractProject and ProjectTracker track project references using the project bin output path.
-            // Setting the command line arguments should have already set the output file name and folder.
-            // We fetch this output path to add the reference.
-            var referencedProject = this.ProjectTracker.GetProject(abstractProject.Id);
-            var binPathOpt = referencedProject.BinOutputPath;
-            if (!string.IsNullOrEmpty(binPathOpt))
-            {
-                AddMetadataReferenceAndTryConvertingToProjectReferenceIfPossible(binPathOpt, properties);
-            }
+            AddProjectReference(new ProjectReference(abstractProject.Id, properties.Aliases, properties.EmbedInteropTypes));
```

Comment:
I think I was hitting an assert or an exception during AbstractProject.Dispose in one of our unit tests. I think different ordering of Dispose calls [here](http://source.roslyn.io/#Roslyn.VisualStudio.CSharp.UnitTests/ProjectSystemShim/CPS/CSharpReferencesTests.cs,66) with your change might cause it to repro again - probably try moving project3.Dispose() upfront.